### PR TITLE
Si/feat/posts

### PR DESCRIPTION
**Pull request by siqbal181**
Couldn't figure out the infinite scroll quirk on your branch, so I think just need to remove this for now :(

### CODE REVIEW

File: frontend/src/components/PostItem/PostItem.css [hunk 1]
```diff
@@ -1,7 +1,13 @@
+.post-container {
+  padding-bottom: 0.6rem;
+  padding-top: 0.3rem;
+}
+
 .post-title {
   font-size: 1.4rem;
   font-weight: 500;
   margin: 0.2rem auto 0.4rem;
+  color: #05473d;
```
Comment (pablisch) [inline frontend/src/components/PostItem/PostItem.css:10]: Can we keep colours as variables in index.css as much as is practical.

File: frontend/src/components/PostItem/PostItem.tsx [hunk 2]
```diff
@@ -40,72 +42,117 @@ const PostItem: React.FC<PostObject> = ({
...
-        {comments?.length > 0 && <h4>Comments:</h4>}
-        {comments?.map((comment, index) => (
-          <div key={index}>
-            <div className="user-and-avatar">
-            <p>{comment.commentText} - {comment.commentAuthor.username}</p>
-              <img src={comment.commentAuthor.avatar} alt="avatar" className="user-avatar" />
+      <p className="post-main-tags">{tags.mainTags}</p>
+      <p className="post-sub-tag">{tags.subTag}</p>
+      <div className="post">
+        <h2 className="post-title">{postText.title}</h2>
```
Comment (pablisch) [inline frontend/src/components/PostItem/PostItem.tsx:78]: title and body are duplicated here.

**Review by pablisch — CHANGES_REQUESTED** [review]
Fine with specific colours being in fine but if we are likely to use them elsewhere, it would be good to get them as variables.

**Comment by pablisch** [comment]
Feels like there is way too much going on in this branch. I've looked at the infinite scroll and all looks good so I can't see why it is not working, though of course I have no idea what makes infinite scroll works but feels like what was working should have been merged in before infinite scroll started as it is a whole new parade.

When I got onto the branch, I got a load of errors and started adding stuff to fix them before I realised I probably needed to npm install instead. I added some logs for visibility and it is all on a new branch, sipj/feat/infinite-scroll, it does not really add much of any value but jump on it for visibility if you like.

The main thing is that looking at the blog, it looks like you've done everything right. Some changes from the blog code but they all felt like improvements. Perhaps the improvements are the problem? I think my next step would be to create a sample project using infinite scroll to see if it works in isolation. It could be that our existing code is interfering, especially the scroll CSS we already have.

**Comment by pablisch** [comment]
There are also now 13 posts in the test db if that helps for infinite testing.

BTW, I am sure you are already aware but the get more posts function works but is never called by infinite scroll.

File: frontend/src/components/FeedColumn/FeedColumn.tsx [hunk 3]
```diff
@@ -3,48 +3,78 @@ import PostItem from "../PostItem/PostItem";
...
-      if (response.status !== 200) {
-        throw new Error("Error in network response");
-      }
-      const posts = response.data;
-      setFeedPosts(posts);
-    } catch (error) {
-      console.error("Error fetching posts:", error);
-    }
+    axios
+      .get("http://localhost:3000/api/v1.0/posts", {
```
Comment (siqbal181) [inline frontend/src/components/FeedColumn/FeedColumn.tsx:14]: stick to prev version of async await

**Pull request merged**